src/main/java/seedu/duke/shooter/MediumSkill.java

```diff
@@ -0,0 +1,62 @@
+package seedu.duke.shooter;
+
+import seedu.duke.Formatter;
+import seedu.duke.ui.Ui;
+
+public class MediumSkill extends Shooter {
+    private String name;
+    private int power = 1;
+    private final int skill = 2;
+
+    public MediumSkill(String name) {
+        super(name);
+    }
+
+    @Override
+    public void printSelfInfo() {
+        super.printSelfInfo();
+    }
+
+    @Override
+    protected void printPower() {
+        System.out.print("Power:");
+        for (int i=0; i<3;i++){
+            if (i<power){
+                System.out.print(" 000");
+            }else{
+                System.out.print(" OOO");
+            }
+        }
+        if (power==1){
+            System.out.println(" Level-Beginner");
+        }else if (power == 2){
+            System.out.println(" Level-Medium");
+        }else{
+            System.out.println(" Level-Expert");
+        }
+    }
+
+    @Override
+    protected void printSkill(){
+        System.out.print("Skill:");
+        for (int i=0; i<3;i++){
+            if (i<skill){
+                System.out.print(" 000");
+            }else{
+                System.out.print(" OOO");
+            }
+        }
+        System.out.println(" Level-Medium");
+    }
+
+    @Override
+    public void printGoalBeforeShoot() {
+        Formatter.printGoalBeforeShotforMedium(Ui.roundCount);
```

Comment:
I moved roundCount to the MatchStat class. You may need to change here after my merge.

Reply:
Ok